### PR TITLE
Bluetooth: Controller: Fix Periodic Advertising to setup Power Amp 

### PR DESCRIPTION
**Pull request by cvinayak**
Fix Periodic Advertising implementation to setup Power
Amplifier (PA) GPIO toggle for transmission instead of
incorrect Low Noise Amplifier (LNA) setup which is for
reception.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by ppryga-nordic — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c [hunk 1]
```diff
@@ -217,6 +217,14 @@ static int prepare_cb(struct lll_prepare_param *p)
 	remainder = p->remainder;
 	start_us = radio_tmr_start(1, ticks_at_start, remainder);
 
+#if defined(CONFIG_BT_CTLR_PROFILE_ISR) || \
+	defined(HAL_RADIO_GPIO_HAVE_PA_PIN)
+	/* capture end of AUX_SYNC_IND/AUX_CHAIN_IND PDU, used for calculating
+	 * next PDU timestamp.
+	 */
+	radio_tmr_end_capture();
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c:225]: Why the PDU end timer capture is enabled here? And why it is conditionally compilated?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c:225]: - In Periodic Advertising without chaining there was no use for LLL to get the end time from radio, hence there was no call to `radio_tmr_end_capture` to capture the radio end time
- With chaining the `sw_switch` used PPI/DPPI for back to back Tx, no radio end time capture was needed there either
- For PA LNA (and ISR profiling), the radio end time is required to setup the GPIOTE using `radio_gpio_pa_lna_enable` which needs call to `radio_tmr_tifs_base_get`, both PA/LNA and ISR profiling call `radio_tmr_end_get`

Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c:225]: Could you add this information to comment above the `radio_tmr_end_capture` call? This is something that helps a lot to understand why the call is there and why it is conditionally compiled.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c [hunk 2]
```diff
@@ -223,6 +217,14 @@ static int prepare_cb(struct lll_prepare_param *p)
 	remainder = p->remainder;
 	start_us = radio_tmr_start(1, ticks_at_start, remainder);
 
+#if defined(CONFIG_BT_CTLR_PROFILE_ISR) || \
+	defined(HAL_RADIO_GPIO_HAVE_PA_PIN)
+	/* capture end of AUX_SYNC_IND/AUX_CHAIN_IND PDU, used for calculating
+	 * next PDU timestamp.
+	 */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c:224]:
```suggestion
    In Periodic Advertising without chaining there was no use for LLL to get the end time from radio, hence there was no call to radio_tmr_end_capture to capture the radio end time
    With chaining the sw_switch used PPI/DPPI for back to back Tx, no radio end time capture was needed there either
    For PA LNA (and ISR profiling), the radio end time is required to setup the GPIOTE using radio_gpio_pa_lna_enable which needs call to radio_tmr_tifs_base_get, both PA/LNA and ISR profiling call radio_tmr_end_get


	 */
```

**Review by ppryga-nordic — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c [hunk 3]
```diff
@@ -223,6 +217,26 @@ static int prepare_cb(struct lll_prepare_param *p)
...
+	 * radio_tmr_end_capture() to capture the radio end time.
+	 *
+	 * With chaining the sw_switch used PPI/DPPI for back to back Tx, no
+	 * radio end time capture was needed there either.
+	 *
+	 * For PA LNA (and ISR profiling), the radio end time is required to
+	 * setup the GPIOTE using radio_gpio_pa_lna_enable which needs call to
+	 * radio_tmr_tifs_base_get(), both PA/LNA and ISR profiling call
+	 * radio_tmr_end_get().
+	 */
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_sync.c:236]:
```suggestion
	 * In Periodic Advertising without chaining there is no need for LLL to
	 * get the end time from radio, hence there is no call to
	 * radio_tmr_end_capture() to capture the radio end time.
	 *
	 * With chaining the sw_switch used PPI/DPPI for back to back Tx, no
	 * radio end time capture is needed there either.
	 *
	 * For PA LNA (and ISR profiling), the radio end time is required to
	 * setup the GPIOTE using radio_gpio_pa_lna_enable which needs call to
	 * radio_tmr_tifs_base_get(), both PA/LNA and ISR profiling call
	 * radio_tmr_end_get().
	 */
```

**Review by Thalley — CHANGES_REQUESTED** [review]
A few minor suggestions regarding using `is` vs `was`, but the main request is for the commit message: 
```
Bluetooth: Controller: Fix Periodic Advertising PA implementation
```
Is too generic for my taste, and should be more specific, i.e. what does this fix?

**Review by Thalley — APPROVED** [review]

**Pull request merged**